pkg/actions/cluster/delete_test.go

```diff
@@ -0,0 +1,127 @@
+package cluster
```

Comment:
mhmm does this test file actually get run when you run `ginkgo -r pkg/action/cluster/`? I would of thought it needs to be in `cluster_test` suite to register. But I get that it needs to have access to the private function `drainAllNodeGroups`.

I'd prefer if we made `drainAllNodeGroups` public, and moved this file into `cluster_test` package.

what do you think :smile: ?

Reply:
This was my initial problem, which caused me to move it to a different package.
I didn't want to make the `drainAllNodeGroups` public so I moved it to be in the same package.
Do you think it is a good thing to make it publicly?

Reply:
He means, that he prefers making the function public, and then testing it, rather than having the test code in the same package as the code it self, aka, the test code is in `cluster` package rather than `cluster_test` package which makes it a separate package. The benefit of having `cluster_test` is that you test the public API of the package. If a private function can't be tested through that, it means that it would need a bit of refactoring maybe or it's not working encapsulating, etc.

This isn't a dogma of course. :) It's a suggestion. Try and test private functions through utilising the public function. If it's hard to test, maybe there are certain conditions which are difficult to achieve in the first place?

Reply:
well summarised @Skarlso . Does that make sense @AmitBenAmi ? Its not a blocker on merging as its a personal preference